include/c3d/base.h

```diff
@@ -1,41 +1,152 @@
+/**
+ * @file base.h
+ * @brief Base citro3d functions
+ */
 #pragma once
 #include "buffers.h"
 #include "maths.h"
 
+/// Default command buffer size
 #define C3D_DEFAULT_CMDBUF_SIZE 0x40000
 
+/**
+ * @brief Data type of indices for use with \ref C3D_DrawElements().
+ */
 enum
 {
-	C3D_UNSIGNED_BYTE = 0,
-	C3D_UNSIGNED_SHORT = 1,
+	C3D_UNSIGNED_BYTE = 0,  ///< Unsigned 8-bit interger
+	C3D_UNSIGNED_SHORT = 1, ///< Unsigned 16-bit interger
 };
 
+/**
+ * @brief Initializes citro3d.
+ * @param[in] cmdBufSize Desired size of GPU command buffer. \ref C3D_DEFAULT_CMDBUF_SIZE should generally be used here.
+ * @note If you have a particularly complex scene you might need to specify a larger \p cmdBufSize.
+ *       Conversely, you may want to decrease it if you're particularly concerned about memory consumption.
+ * @return true if library was initialized successfully, false if there was an error.
+ */
 bool C3D_Init(size_t cmdBufSize);
+
+/**
+ * @brief Deinitializes citro3d
+ * @sa C3D_Init()
+ */
 void C3D_Fini(void);
 
+/**
+ * @brief Retrieves the current command buffer usage.
+ * @return Fraction of command buffer used. (0.0f to 1.0f)
+ */
 float C3D_GetCmdBufUsage(void);
 
+/**
+ * @brief Binds a shader program to the current rendering state.
+ * @param[in] program Specifies the pointer to a shader program object whose executables are to be used
+ *       as part of the current rendering state.
+ */
 void C3D_BindProgram(shaderProgram_s* program);
 
+/**
+ * @brief Sets the viewport for the current framebuffer.
+ * @note This value is modified by \ref C3D_FrameDrawOn(). (using values specified by \ref C3D_RenderTargetSetOutput())
+ * @note When using this with a rendertarget intended for display, keep in mind the orientation of the screens.
```

Comment:
Is this viewport aligned with the screens (i.e. "width" is the narrower dimension, since the screens are in portrait mode)? Might be good to clarify

Reply:
Meh, if you are calling this function, you should probably already know that.
That being said `\ref C3D_RenderTargetSetOutput()` should be `C3D_RenderTargetCreate()` and I apparently forgot to add a note to `C3D_RenderTargetCreate()`.